plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerConnectorTest.java

```diff
@@ -640,6 +642,209 @@ protected void verifyColumnNameLengthFailurePermissible(Throwable e)
+                    "The statement did not return a result set.");
+        }
+    }
+
+    @Test
+    public void testProcedureWithIfElseStatement()
+    {
+        try (TestProcedure testProcedure = createTestingProcedure(
+                "@id INTEGER",
+                """
+                IF @id > 50
+                    SELECT 1 as first_column;
+                ELSE
+                    SELECT '2' as second_column;
+                """)) {
+            assertQueryFails(
+                    "SELECT * FROM TABLE(system.procedure(schema => '%s', procedure => '%s', inputs => ARRAY['100'])) "
+                            .formatted(getSession().getSchema().orElseThrow(), testProcedure.getName()),
+                    "Procedure has multiple ResultSets for query: .*");
+        }
+    }
+
+    @Test
+    public void testProcedureWithSingleIfElse()
+    {
+        try (TestProcedure testProcedure = createTestingProcedure("SELECT 1 as first_row; SELECT 2 as second_row")) {
+            assertQueryFails(
+                    "SELECT * FROM TABLE(system.procedure(schema => '%s', procedure => '%s')) "
+                            .formatted(getSession().getSchema().orElseThrow(), testProcedure.getName()),
+                    "Procedure has multiple ResultSets for query: .*");
+        }
+    }
+
+    @Test
+    public void testProcedureWithMultipleResultSet()
+    {
+        try (TestProcedure testProcedure = createTestingProcedure("SELECT 1 as first_row; SELECT 2 as second_row")) {
+            assertQueryFails(
+                    "SELECT * FROM TABLE(system.procedure(schema => '%s', procedure => '%s')) "
+                            .formatted(getSession().getSchema().orElseThrow(), testProcedure.getName()),
+                    "Procedure has multiple ResultSets for query: .*");
+        }
+    }
+
+    @Test
+    public void testProcedureWithCreateOperation()
+    {
+        String tableName = "table_to_create" + randomNameSuffix();
+        try (TestProcedure testProcedure = createTestingProcedure("CREATE TABLE %s (id BIGINT)".formatted(tableName))) {
+            assertQueryFails(
+                    "SELECT * FROM TABLE(system.procedure(schema => '%s', procedure => '%s'))"
+                            .formatted(getSession().getSchema().orElseThrow(), testProcedure.getName()),
+                    "Failed to get table handle for procedure query. The statement did not return a result set.");
+            assertQueryReturnsEmptyResult("SHOW TABLES LIKE '%s'".formatted(tableName));
+        }
+    }
+
+    @Test
+    public void testProcedureWithDropOperation()
+    {
+        try (TestTable table = new TestTable(onRemoteDatabase(), "table_to_drop", "(id BIGINT)")) {
+            try (TestProcedure testProcedure = createTestingProcedure("DROP TABLE " + table.getName())) {
```

Comment:
https://learn.microsoft.com/en-us/sql/t-sql/queries/output-clause-transact-sql?view=sql-server-ver16 exists so can we test with it as well?

Reply:
Have added test with `OUTPUT` clause.